twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -267,6 +280,14 @@ void stopByIdAndWait(String runnableName, int instanceId) {
 
         resourceReport.removeRunnableResources(runnableName, containerId);
         containerChange.signalAll();
+        if (eventHandler != null) {
+          Integer exitStatus = containerExitStatus.get(containerId);
+          if (exitStatus == null) {
```

Comment:
So this is for handling the case when `handleCompleted` was not called (e.g. the runnable container is not stopping after receiving the "stop" command), right? If that's the case, isn't that the exit status is always not there?

Also, do we need to do the similar logic to call event handler in the `stopAll` method?